### ISSUE TITLE
Nlmgs containing Tcmsg fails serialization

### ISSUE THREAD

**Issue by tammela** [jbaublitz/neli#152]
**Version of neli**
0.5.3

**Does the documentation specify that neil's behavior is correct?**
Not related.

**Does there appear to be a mistake in the netlink documentation based on the kernel code?**
Not related.

**Describe the bug**
Serialization of Tcmsgs seems to be bugged.
It's reporting `The number of bytes written to the buffer did not fill the given space` for very simple messages.

In an example, I have noticed that the buffer is expecting 40 bytes to be written as it's aligned to 4 bytes.
The message however is reporting 39 bytes, perhaps it's missing some padding?

**To Reproduce**
```
        let msg = Tcmsg {
            tcm_family: 0,
            tcm_ifindex: 0,
            tcm_handle: 0,
            tcm_parent: 0,
            tcm_info: 0,
            rtattrs: RtBuffer::new(),
            padding_char: 0,
            padding_short: 0,
        };
        
    let mut socket = NlSocketHandle::connect(socket::NlFamily::Route, None, &[])?;
    let nlhdr = {
        let nl_type = Rtm::Newtfilter;
        let nl_flags = nl::NlmFFlags::new(&[nl::NlmF::Request, nl::NlmF::Ack, nl::NlmF::Excl, nl::NlmF::Create]);
        let payload = NlPayload::Payload(msg);
        Nlmsghdr::new(None, nl_type, nl_flags, None, None, payload)
    };

    println!("{:?}", nlhdr);

    socket.send(nlhdr)?;
```

**Additional context**
I modified the library to expose the padding fields so I could build the Tcmsg payload.
See https://github.com/jbaublitz/neli/issues/151.

**Comment by jbaublitz** [comment]
@tammela Can you please take a look and see if this is still a problem in v0.6.0? I've heavily improved the serialization code. If it hasn't, I'll fix it and add some tests.

**Comment by jbaublitz** [comment]
Considering this closed by 16854b52d52fefed8e3ce3337b58499d23723934
